src/components/Routes.jsx

```diff
@@ -0,0 +1,87 @@
+import { Route, Switch } from 'react-router-dom';
+import UsersWithContext from './ContextHOCs/UsersWithContext';
+import Tasks from '../pages/Tasks';
+import UsersTasksWithContext from './ContextHOCs/UsersTasksWithContext';
+import UsersTracks from '../pages/UsersTracks';
+import UsersProgress from '../pages/UsersProgress';
+import PageNotFound from '../pages/PageNotFound';
+import settingsIcon from '../icons/settings.svg';
+import usersIcon from '../icons/user.svg';
+import tasksIcon from '../icons/tasks.svg';
+import MenuItem from './Aside/Menu/MenuItem/MenuItem';
+import SettingsWithContext from './ContextHOCs/SettingsWithContext';
+
+const paths = [
+  { isExact: true, path: '/main/users', component: UsersWithContext },
+  { isExact: true, path: '/main/tasks', component: Tasks },
+  { isExact: true, path: '/main/users/:userId/tasks', component: UsersTasksWithContext },
+  { isExact: true, path: '/main/users/:userId/tasks/:taskId/track', component: UsersTracks },
+  { isExact: true, path: '/main/users/:userId/progress', component: UsersProgress },
+  { isExact: true, path: '/main/settings', component: SettingsWithContext },
+  { isExact: false, path: '', component: PageNotFound },
+];
+
+const adminAndMentorPermissions = [0, 1, 2, 3, 4, 5, 6];
+const userPermissions = [2, 3, 5, 6];
+
+export const getRoleDependedRoutes = (loggedUser) => {
+  let permissionsList = null;
+  if (loggedUser.role === 'Admin' || loggedUser.role === 'Mentor') {
+    permissionsList = adminAndMentorPermissions;
+  } else if (loggedUser.role === 'User') {
+    permissionsList = userPermissions;
+  }
+
+  const routePack = permissionsList.map((item) => {
+    return (
+      <Route
+        key={item.toString()}
+        exact={paths[item].isExact}
+        path={paths[item].path}
+        component={paths[item].component}
+      />
+    );
+  });
+
+  return <Switch>{routePack}</Switch>;
+};
+
+const menuList = [
+  {
+    title: 'Users',
+    path: '/main/users',
+    image: usersIcon,
+  },
+  {
+    title: 'Workflow',
+    path: '/main/tasks',
+    image: tasksIcon,
+  },
+  {
+    title: 'Settings',
+    path: '/main/settings',
+    image: settingsIcon,
+  },
+];
+
+const adminAndMentorMenuList = [0, 1, 2];
+const userMenuList = [2];
+
+export const getRoleDependedMenuLinks = (loggedUser) => {
+  const isUser = loggedUser.role === 'User';
+  let roleDependendMenuList = null;
+  if (!isUser) {
+    roleDependendMenuList = adminAndMentorMenuList;
+  } else if (isUser) {
+    roleDependendMenuList = userMenuList;
+  }
```

Comment:
you can re-write it with ternary operator to more short and nice code

Reply:
it was perfectly updated:
```suggestion
export const getRoleDependedMenuLinks = ({ role }) => {
  const rolePermissionsLists = menuList.filter((menuObj) => menuObj.permissions.includes(role));

  return rolePermissionsLists.map((menuObj, index) => (
    <MenuItem key={index.toString()} title={menuObj.title} path={menuObj.path} image={menuObj.image} />
  ));
};
```